mac

```diff
@@ -233,10 +259,17 @@ if ! brew_is_installed "python3"; then
   brew 'pyenv-virtualenv'
   brew 'pyenv-virtualenvwrapper'
 EOF
-  # shellcheck disable=SC2016
-  append_to_file "$shell_file" 'if which pyenv > /dev/null; then eval "$(pyenv init -)"; fi'
-  # shellcheck disable=SC2016
-  append_to_file "$shell_file" 'if which pyenv-virtualenv-init > /dev/null; then eval "$(pyenv virtualenv-init -)"; fi'
+  case "$SHELL" in
+    */fish) :
+      append_to_file "$shell_file" "status --is-interactive; and source (pyenv init -|psub)"
+      append_to_file "$shell_file" "status --is-interactive; and source (pyenv virtualenv-init -|psub)"
```

Comment:
It might be useful to have a short comment here reminding that you can't use `append_to_posix_shell_file` for fish because that function doesn't do anything in fish shells (I thought about asking why not use that function, but when I scrolled up to look at the value of `$shell_file` I saw and remembered 😛)